storage/fcds/fcds.go

```diff
@@ -117,18 +119,24 @@ func (s *Store) Get(addr chunk.Address) (ch chunk.Chunk, err error) {
 		return nil, err
 	}
 
-	sh := s.shards[getShard(addr)]
+	sh := s.shards[m.Shard]
 	sh.mu.Lock()
-	defer sh.mu.Unlock()
```

Comment:
Why is this defer removed?

Reply:
the idea was not to hold the lock beyond the specific time that the shard is used (i.e. to not hold it for the metric increment or the chunk constructor)

Reply:
i will put it back as a defer statement